pkg/e2e/infra/infra.go

```diff
@@ -195,9 +195,22 @@ var _ = g.Describe("[Feature:Machines] Managed cluster should", func() {
 		err = waitForClusterSizeToBeHealthy(client, initialClusterSize)
 		o.Expect(err).NotTo(o.HaveOccurred())
 
-		workerNode, err := getWorkerNode(client)
+		g.By("getting worker node")
+		var workerNode *corev1.Node
+		o.Eventually(func() bool {
+			workerNodes, err := e2e.GetWorkerNodes(client)
+			if err != nil {
+				return false
+			}
+			if len(workerNodes) == 0 {
```

Comment:
Is this fixing a known issue? If not, I would still prefer to use the original form `workerNode, err := e2e.GetWorkerNodes(client)`. If we want to assert the fact that there is at least one worker machine, it needs to go outside of `o.Eventually` so the test can fail in case there is no worker node.

Reply:
I just saw that `getWorkerNode` used `wait.PollImmediate`, so I entered the `e2e.GetWorkerNodes(client)` under the `Eventually`, but if `wait.PollImmediate` was some redundant code I will happily remove the `Eventually`.